From be0239d8873fe9ef283a8f7c738fd83ed1974011 Mon Sep 17 00:00:00 2001
From: Mark Enriquez <enriquezmark36@gmail.com>
Date: Thu, 26 Sep 2019 21:22:37 +0800
Subject: [PATCH 15/15] gralloc: scx15: Expose USE_3_FRAMEBUFFER flag

On your BoardConfig.mk, set USE_3_FRAMEBUFFER to true.

This makes the gralloc module interrogate the FB (framebufer) to use
a contiguous display buffer the size of three displays.

Usually this is not needed, but on Android 7 and above,
the failure to query for the display parameters not only
disables page-flipping but also borks the dpi parameters.
This will cause some apps that depend on screen dpi,
for scaling, to display ant size images or elements.

Change-Id: Ie060bb73c1776f5fecff3ab487094c97b11b8a5f
Signed-off-by: Mark Enriquez <enriquezmark36@gmail.com>
---
 gralloc/scx15/Android.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gralloc/scx15/Android.mk b/gralloc/scx15/Android.mk
index 718fa89..6baf568 100644
--- a/gralloc/scx15/Android.mk
+++ b/gralloc/scx15/Android.mk
@@ -81,6 +81,10 @@ LOCAL_CFLAGS += -DHIDL_INVALID_FD
 LOCAL_CFLAGS += -DHIDL_NO_FREE_FB
 endif
 
+ifeq ($(TARGET_USE_3_FRAMEBUFFER), true)
+LOCAL_CFLAGS += -DUSE_3_FRAMEBUFFER
+endif
+
 LOCAL_SRC_FILES := \
 	gralloc_module.cpp \
 	alloc_device.cpp \
-- 
2.23.0

